File: app/src/main/java/nl/acidcats/teatimer/ui/SplashActivity.java
package nl.acidcats.teatimer.ui;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import nl.acidcats.teatimer.alarm.AlarmHelper;
import nl.acidcats.teatimer.util.AppShortcutUtil;
import nl.acidcats.teatimer.util.BundleUtil;

/**
 * Created on 23/11/2017.
 */

public class SplashActivity extends AppCompatActivity {

    private static final int DEFAULT_TEA_TIME_MINS = 5;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);


        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
            AppShortcutUtil.setupShortcuts(this);
        }

        if (AlarmHelper.isAlarmRunning()) {
            startActivity(new Intent(this, TeaTimeActivity.class));
        } else {
            startAlarm();
        }

        finish();
    }

    private void startAlarm() {
        int timerMins = DEFAULT_TEA_TIME_MINS;
        if (getIntent() != null) {
            timerMins = BundleUtil.getBundleValue(getIntent().getExtras(), AppShortcutUtil.KEY_TIME, DEFAULT_TEA_TIME_MINS);
        }

        AlarmHelper.startAlarm(this, timerMins, TeaTimeActivity.class);
    }

}

File: app/src/main/java/nl/acidcats/teatimer/ui/SplashActivity.kt
package nl.acidcats.teatimer.ui

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.github.ajalt.timberkt.Timber
import nl.acidcats.teatimer.R
import nl.acidcats.teatimer.TimerActionReceiver
import nl.acidcats.teatimer.helpers.AppShortcutHelper
import nl.acidcats.teatimer.helpers.ConfigHelper
import nl.acidcats.teatimer.helpers.StorageHelper
import nl.acidcats.teatimer.util.IntentKey
import nl.acidcats.teatimer.util.getIntExtra
import org.koin.android.ext.android.inject

/**
 * Created on 23/11/2017.
 */
@SuppressLint("CustomSplashScreen")
class SplashActivity : AppCompatActivity() {

    private val storageHelper: StorageHelper by inject()
    private val appShortcutHelper: AppShortcutHelper by inject()
    private val configHelper: ConfigHelper by inject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        appShortcutHelper.setupShortcuts()

        if (storageHelper.alarmState.isAlarmRunning) {
            startActivity(Intent(this, MainActivity::class.java))
        } else {
            val duration = intent.getIntExtra(IntentKey.Duration, configHelper.defaultDuration)
            Timber.d { "duration = $duration" }

            applicationContext.sendBroadcast(
                TimerActionReceiver.getStartIntent(
                    context = applicationContext,
                    duration = duration
                )
            )
        }

        finish()
    }
}
